src/webServer/webServer.ts

```diff
@@ -130,10 +130,10 @@ namespace ts.server {
             getCurrentDirectory: returnEmptyString, // For inferred project root if projectRoot path is not set, normalizing the paths
 
             /* eslint-disable no-restricted-globals */
-            setTimeout,
-            clearTimeout,
+            setTimeout: (cb, ms, args) => setTimeout(cb, ms, args),

```

Comment:
The signature has _rest args_ for the 3rd–Nth arguments because those arguments work like `.bind` and partially apply `cb`:

<img width="408" alt="image" src="https://user-images.githubusercontent.com/3277153/127686571-266fa048-67a3-4869-8746-3f762792ad5d.png">

_Hopefully_ nobody’s using them because that’s a horrible and wholly unnecessary pattern, but for the current definition of `setTimeout`, without the rest arg, this _could_ be breaking. Maybe it would be worthwhile just to change the signature to not allow those extra arguments.

Reply:
I think you're suggesting I change `(cb, ms, args)` to `(cb, ms)`, but I got a little lost.  Please confirm?

Reply:
Here’s the type you’re implementing:

```ts
export interface ServerHost extends System {
  // ...
  setTimeout(callback: (...args: any[]) => void, ms: number, ...args: any[]): any;
}
```

Note that it’s variadic. Your implementation only captures the _first_ of its rest params. So someone is allowed to call

```ts
serverHost.setTimeout((a, b, c) => /* do something with a, b, and c */, 100, "a", "b", "c")
```

and they would expect their callback to be called with arguments `"a"`, `"b"`, and `"c"`. But with the implementation you have here, their function would be called only with `"a"`, leaving `b` and `c` undefined.

So I’m suggesting that you should _either_ fully implement the signature as written by using rest parameters:

```diff
- setTimeout: (cb, ms, args) => setTimeout(cb, ms, args),
+ setTimeout: (cb, ms, ...args) => setTimeout(cb, ms, ...args),
```

_or_ change the definition of `ServerHost['setTimeout']` so callers can’t pass extra args:


```diff
export interface ServerHost extends System {
  // ...
-   setTimeout(callback: (...args: any[]) => void, ms: number, ...args: any[]): any;
+   setTimeout(callback: (...args: any[]) => void, ms: number): any;
}
```

which would mean you would also then drop `args` off your implementation:

```diff
- setTimeout: (cb, ms, args) => setTimeout(cb, ms, args),
+ setTimeout: (cb, ms) => setTimeout(cb, ms),
```

I think doing the latter is fine, because the ability to pass those extra args is completely unnecessary, and it’s not implemented in a type-safe way, so it seems unlikely that anyone is relying on that behavior. But it’s technically a breaking API change.

Reply:
Alternatively, you can switch to `setTimeout: setTimeout.bind(globalThis)` or `setTimeout: setTimeout.bind(self)` or `setTimeout: setTimeout.bind(window)` if you want to do a little research as to which one(s) are correct for the places this host needs to run.

Reply:
For some reason, I got it into my head that the dots were redundant.  Oops.